src/Barcode.php

```
@@ -62,7 +62,7 @@ public function __construct($options = null)
             if (array_key_exists('options', $options)) {
                 $options['options'] = ['options' => $options['options']];
             }
-        } elseif ($options instanceof Traversable) {
+        } elseif ($options instanceof ArrayAccess) {
```

I think this is the only controversial change here. To be honest, anyone _not_ passing in something that implements ArrayAccess here would have failed on line 67 anyhow

---

Not necessary. `property_exists` in next check is weird too. It does not use either `__isset` or `offsetExists`.
So both those checks are wrong.

Bug was introduced here with a wrong fix:
https://github.com/laminas/laminas-validator/pull/32

---

Correct fix here is first check for `Traversable` and apply `iterator_to_array`. Then do check for array.
This was the common approach for options. Check if abstract validator and other validators do it that way.

Also adapter option here is not checked at all. any garbage passed here would be silently replaced in `getAdapter()`. yikes

---

Correct fix here is first check for `Traversable` and apply `iterator_to_array`. Then do check for array.
This was the common approach for options. Check if abstract validator and other validators do it that way.

Also adapter option here is not checked at all. any garbage passed here would be silently replaced in `getAdapter()`. yikes

---

I think it would be better if handled in a separate PR

---

Thanks @Xerkus - I've also corrected a whole load of types in the mucky options arrays for Barcode and barcode adapter - annotations were a fair distance away from reality.